### PR TITLE
Vite

### PR DESCRIPTION
**Pull request by jgw96**
This PR:
- Updates our build setup to use https://vitejs.dev/ along with needed files and updated folder structure
- Removes unnecessary files to simplify things
- Updates the package.json to use https://vite
- Updated our dependences as Vite does alot of what we were doing custom

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -20,18 +20,14 @@
...
-    <link rel="icon" href="assets/icons/icon_24.png" type="image/x-icon" />
+    <link rel="icon" href="icons/icon_24.png" type="image/x-icon" />
 
     <!-- Imports the manifest to represent the web application. A web app must have a manifest to be a PWA. -->
     <link rel="manifest" href="manifest.json" />
 
-    <!-- A stylesheet to design the pages. -->
-    <link href="styles/global.css" rel="stylesheet" />
-
     <script type="module" src="https://unpkg.com/@fluentui/web-components"></script>
```
Comment (nmetulev) [inline index.html:28]: Can we import the components as needed es modules instead of binging in all components?

File: package.json [hunk 2]
```diff
@@ -4,45 +4,34 @@
   "description": "A starter kit for building PWAs!",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "build": "rimraf dist/ && rollup --config rollup.config.js && node workbox.mjs",
-    "start": "concurrently \"rollup --config rollup.config.dev.js --watch\" \"web-dev-server --app-index build/index.html --root-dir build/ --compatibility none --node-resolve --watch --open /\"",
-    "dev": "concurrently \"rollup --config rollup.config.dev.js --watch\" \"web-dev-server --app-index build/index.html --root-dir build/ --compatibility none --node-resolve --watch --open /\"",
-    "dev-task": "concurrently \"rollup --config rollup.config.dev.js --watch\" \"web-dev-server --app-index build/index.html --root-dir build/ --compatibility none --node-resolve --watch /\""
+    "dev-task": "vite",
```
Comment (nmetulev) [inline package.json:7]: consider calling this 'dev' instead for folks who are used to 'npm run dev'

File: index.html [hunk 3]
```diff
@@ -20,18 +20,14 @@
     <meta name="apple-mobile-web-app-status-bar-style" content="black" />
 
     <!-- Imports an icon to represent the document. -->
-    <link rel="icon" href="assets/icons/icon_24.png" type="image/x-icon" />
+    <link rel="icon" href="icons/icon_24.png" type="image/x-icon" />
```
Comment (nmetulev) [inline index.html:23]: shouldn't this still be `assets\icons\icon..` since everything is in the assets folder under public?

**Review by nmetulev — COMMENTED** [review]
Amazing!

Couple questions/thoughts?
* Is registerSW.js necessary?
* could .eslintrc.json be moved inside package.json, eliminating another file?

File: index.html (hunk 1 above)
Comment (jgw96) [reply index.html:28]: Yes we could, however I'm not a fan of the developer experience of having to register each component you import. I feel like it easy to forget / miss. I realize it's also a small bit of code in the scheme of things too though and that there are probably even ways PWA Studio could help here. Let me know what you think 😊

Comment (jgw96) [reply index.html:28]: Oh and yes, registerSW.js is necessary, otherwise there is a console error in dev mode

File: package.json (hunk 2 above)
Comment (jgw96) [reply package.json:7]: ahh, I think this was actually changed as a mixup, good catch

File: index.html (hunk 3 above)
Comment (jgw96) [reply index.html:23]: yet another good catch

File: index.html (hunk 1 above)
Comment (nmetulev) [reply index.html:28]: > Yes we could, however I'm not a fan of the developer experience of having to register each component you import. I feel like it easy to forget / miss. I realize it's also a small bit of code in the scheme of things too though and that there are probably even ways PWA Studio could help here. Let me know what you think 😊

Yeah, I know what you mean, I'm not a fan of registering each individual component either. I guess it's a trade of between convenience and performance. I wish there was a way we could do build time analysis and auto import the needed components.

**Review by nmetulev — APPROVED** [review]

**Pull request merged**